### PR TITLE
wrapper counts batch bytes

### PR DESCRIPTION
**Pull request by sasha8**
### Description
Hopefully, all the logic to start testing is there. 
A few optimizations and refactoring in TODOs

### CODE REVIEW

File: consensus/src/quorum_store/quorum_store_wrapper.rs [hunk 1]
```diff
@@ -93,20 +97,40 @@ impl QuorumStoreWrapper {
...
-                sequence_number: txn.sequence_number(),
-            }));
 
-        // TODO: that is a bug! QuorumStore counts in bytes - need to keep track of it.
-        // TODO: also some timer if there are not enough txns
-        if self.batch_in_progress.len() <= self.max_batch_size - MAX_SUB_BATCH_SIZE {
+        let mut end_batch = false;
+        let mut txns_data = Vec::new();
+
+        // TODO: pass TxnData to QuorumStore to save extra serialization.
```
Comment (bchocho) [inline consensus/src/quorum_store/quorum_store_wrapper.rs:107]: This serialization is required once to save to QuorumStore, so we are not adding any additional overhead here once the TODO is taken care of?

File: consensus/src/quorum_store/quorum_store_wrapper.rs [hunk 2]
```diff
@@ -93,20 +97,40 @@ impl QuorumStoreWrapper {
...
+                self.batch_in_progress.push(TransactionSummary {
+                    sender: txn.sender(),
+                    sequence_number: txn.sequence_number(),
+                });
+                self.bytes_in_progress = self.bytes_in_progress + bytes.len();
+                let mut hasher = DefaultHasher::new(b"TxnData");
+                hasher.update(&bytes);
+                txns_data.push(TxnData {
+                    txn_bytes: bytes,
+                    hash: hasher.finish(),
```
Comment (bchocho) [inline consensus/src/quorum_store/quorum_store_wrapper.rs:124]: Actually, if we're writing a hash per transaction, isn't this significantly more expensive than the approach of hash per QuorumStoreBatch? The size and also the setup cost of creating a new hasher per txn might be significant.

Comment (sasha8) [reply consensus/src/quorum_store/quorum_store_wrapper.rs:124]: Not sure. We are hashing the same amount of data. Do you think creating an hasher is a lot of overhead? Anyway, the hash is only for the addressable storge Rati wants to implement, so if we don't do we don't need the hashes.

File: consensus/src/quorum_store/quorum_store_wrapper.rs (hunk 1 above)
Comment (sasha8) [reply consensus/src/quorum_store/quorum_store_wrapper.rs:107]: Not for the serialization.

**Pull request merged**